repose-aggregator/components/filters/rate-limiting-filter/src/main/java/org/openrepose/filters/ratelimiting/RateLimitingHandler.java

```diff
@@ -109,29 +115,76 @@ private boolean requestHasExpectedHeaders(HttpServletRequest request) {
         return request.getHeader(PowerApiHeader.USER) != null;
     }
 
+    private static Optional<MediaType> parseMediaType(String type) {
+        Optional<MediaType> parsedType = Optional.empty();
+        try {
+            parsedType = Optional.of(MediaType.parseMediaType(type));
+        } catch (IllegalArgumentException iae) {
+            LOG.warn("Media type could not be parsed: {}", type, iae);
+        }
+        return parsedType;
+    }
+
+    private Optional<MediaType> getPreferredMediaType(List<String> acceptValues) {
+        Optional<MediaType> preferredMediaType = Optional.of(DEFAULT_MEDIA_TYPE);
+        if (!acceptValues.isEmpty()) {
+            /*
+            Parses media types from the "Accept" header (dropping any that cannot be parsed),
+            removes any media types that are not supported by this filter (since they will not be returned anyway),
+            then sorts the remaining media types by specificity (primarily) and quality (secondarily).
+              */
+            List<MediaType> sortedRelevantAcceptTypes = acceptValues.stream()
+                    .map(RateLimitingHandler::parseMediaType)
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
+                    .filter(mediaType -> SUPPORTED_MEDIA_TYPES.stream().anyMatch(mediaType::isCompatibleWith))
+                    .sorted(new CompoundComparator<>(MediaType.SPECIFICITY_COMPARATOR, MediaType.QUALITY_VALUE_COMPARATOR))
```

Comment:
So the difference I'm seeing between the specificity and quality comparators is that the quality comparator checks quality first (then specificity, then parameters, if applicable), whereas the specificity comparator checks specificity first (then quality, then parameters, if applicable).

Prioritizing specificity still makes sense to me, I think. It seems like we don't actually need to use a `CompoundComparator`, but their implementation did, so I did.